### PR TITLE
Expand branch protection checks to admins and required review

### PR DESCRIPTION
**Pull request by foolip**
Neither error is shown by default in report.html, as it's still
unclear if these policies have corner cases where they're a bad
idea. Candidate repos to discuss the tradeoffs with will be taken from
the resulting report.

### CODE REVIEW

**Comment by foolip** [comment]
@dontcallmedom this build on https://github.com/w3c/validate-repos/pull/64 and will require rebasing, but I'm submitting it as a draft for discussion.

First, do you think the bare 'unprotectedbranch' check is still useful, or should I drop that and make it an implicit part of checking for required review?

Second, who would go about changing repos to pass this validation, and who should be consulted before starting to do that?

**Comment by dontcallmedom** [comment]
I think we should keep the `unprotectedbranch` check as a stepping stone - I don't think all groups (esp if we consider the vast number of CGs with very different level of formality) are ready or willing to use the stricter commitment to requiring review, and for these groups, it remains useful to at least know we should encourage them to protect their default branch (which has a low formality cost).

In terms of defining which groups ought to adopt that policy and when, I defer to @plehegar - my assumption is that a big part of it is socializing the benefits, and training people on what this implies in their day to day work. We can start reporting it in the tool, but I wouldn't include it in the default report until the social aspects are clearer.

Once there is buy-in from groups, for groups with staff contacts, I would suggest we first ask staff contacts to take care of getting that adopted and deployed by their groups. For CGs, I think it would have to be more ad-hoc.

File: lib/validator.js [hunk 1]
```diff
@@ -159,10 +143,29 @@ function validateRepo(r, licenses, repoData, cgData, repoMap) {
...
+      let rules = [];
+      if (r.branchProtectionRules) {
+        rules = r.branchProtectionRules.nodes.filter(rule => rule.pattern === defaultBranch);
+      }
+      if (rules.length === 0) {
+        reportError('unprotectedbranch', {error: `${defaultBranch} branch is not protected`});
+      } else {
+        rules = rules.filter(rule => rule.requiredApprovingReviewCount > 0);
+        if (rules.length === 0) {
+          reportError('unprotectedbranch', {error: `${defaultBranch} branch review is not required`});
```
Comment (plehegar) [inline lib/validator.js]: LGTM.

**Comment by foolip** [comment]
I'll tweak this to make `unprotectedbranch` be its own error like before, but checking that it's the default branch and also emitting it for `cg-report` repos. Then I'll make the two new errors have their own codes and won't add them to the report yet.

**Comment by foolip** [comment]
@dontcallmedom this is ready for review now, if CI passes...

**Pull request merged**